default.py

```diff
@@ -272,8 +437,33 @@ def play_url(self, url):
             playitem.setProperty('inputstreamaddon', 'inputstream.adaptive')
             playitem.setProperty('inputstream.adaptive.manifest_type', 'hls')
             playitem.setProperty('inputstream.adaptive.stream_headers', url.split('|')[1])
+
+        # Busy dialog until player starts
+
         xbmc.Player().play(item=url, listitem=playitem)
 
+        # Show a busy dialog until playing starts.
+        # Kodi deinits busy dialog on player start.
+        # TODO Make timeout time an add-on variable
+        # Note: Overrides playlisttimeout kodi setting (default 20s)
+        calculatedTimeout = time.time() + 10
+
+        while True:
+            if xbmcgui.getCurrentWindowId() == WINDOW_FULLSCREEN_VIDEO:
+                break
+            elif xbmcgui.getCurrentWindowId() != WINDOW_DIALOG_BUSY:
+                # Needed since inputadaptive disables busy dialog too early
+                show_busy_dialog()
+
+            if time.time() > calculatedTimeout:
+                logger.error('Player took too long to start')
+                break
+
+            # Sleep to avoid hogging CPU
+            xbmc.sleep(100)
```

Comment:
Would be nice to have a comment here noting that this is in ms, especially since Python's time.sleep is in sec.

Not a blocker to merging though. Can be added in later fixes.